### DISCUSSION TITLE
Tagesverbrauch wird nicht im Sunnyportal angezeigt

### DISCUSSION THREAD

**Discussion by Doenerteller** [camueller/SmartApplianceEnabler#398] [Installation und Betrieb]
*** EDIT ***
Für den kurz angebundenen Leser:

Es scheint so zu sein, dass es teilweise ERHEBLICHE zeitliche Verzögerung beim upload der Daten aus dem Home Manager in's Sunnyportal gibt. Es entsteht schnell der Eindruck, dass etwas nicht geht. 

Das Berechnen des Tagesverbrauchs wird anscheinend erst nach dem Vorhandensein einer gewissen Datenmenge angestoßen.
Man muss den Hobel mal mindestens einen Tag laufen lassen und dann gucken.

*** EDIT ***

Servus zusammen!
 
Ich mache gerade notgedrungen die ersten Schritte mit dem SAE, weil SMA ja leider eine Funktionswüste ist, was das plug&play von Verbrauchern angeht. Heutzutage würde ich erwarten, dass man da ein Kabel zieht, die Geräte handeln das Notwendige untereinander aus, und zackbumm ist der Kram im Portal verfügbar. Naja, machste nix...

Mein Ziel ist es meine im März kommende Wärmepumpe zu messen und den Verbrauch darstellen zu lassen. So wie ich das vom EV Charger kenne. Steuern möchte ich nichts, nur messen. Dazu habe ich als Testobjekt meine Lüftungsanlage über einen Shelly Plug S Stecker angeschlossen. Nach den ersten üblichen Bauchschmerzen von so einem mehr oder weniger komplexen Setup habe ich nun den aktuellen Verbrauch im Sunnyportal.

Was mir aber fehlt ist der Tagesverbrauch. Der wird mit Null angezeigt.

![image](https://user-images.githubusercontent.com/119921647/205844023-8481e10b-5710-4cfb-b4eb-b767a2ff30d1.png)

Meine erste Vermutung: das ist zu wenig Gesamtverbrauch, es gibt eine Mindestgrenze im Portal um mehr als Null angezeigt zu bekommen. Das Gerät dümpelte heute mit 10 Watt 7 Stunden lang vor sich hin. Das ist nicht Nichts, aber auch nicht sehr viel.
Andererseits ist die Einheit Wh, da kann man die verbrauchten 70 davon doch ruhig anzeigen.

Hier die Daten, welche der Home Manger sieht

![image](https://user-images.githubusercontent.com/119921647/205845249-e5a87b68-84f6-4bf0-b5c2-8985d8f0de19.png)

Ist daran irgendwas Gurke? Oder muss sich das Ganze erstmal über 24h Einschwingen?

Für Sachdienliche Hinweise wäre ich dankbar. Wenn ich noch irgendwelche Infos posten soll mach ich das gerne, gebt mir
einfach bescheid!

Gruss
Martin

**Comment by Doenerteller** [comment]
Achso...eine weitere Vermutung liegt bei mir in der generellen Schrottigkeit des Home Managers und dessen ruckeliger Zusammenarbeit mit dem Sunnyportal. Ich habe gefühlt nie einen absolut störungsfreien Betrieb des Portals. Irgendein Gerät wird immer als "Ist nicht zu erreichen" angezeigt, obwohl mit meinem Netz und dem Internetzugang alles normal ist.
Auch die SMA App (zieht sich ja auch nur die Daten aus dem Portal) ist ein Graus. Oft wird nichts angezeigt und alles seeehr träge.
Ihr seht schon, ich steht total auf SMA... ;)

![image](https://user-images.githubusercontent.com/119921647/205852232-32de440f-ce20-4dd1-99d3-4954bb88108f.png)

Es ist jetzt 8:50 und die letzten Daten sind von 7:45

![image](https://user-images.githubusercontent.com/119921647/205852543-868e429e-7feb-48ff-8f11-cbbc8b4a6c92.png)

Vielleicht hat es auch einfach damit zu tun, dass die Kommunikation richtung Portal grundsätzlich fragwürdig ist und mit dem SAE ist eigentlich alles in Ordnung?

**Comment by Doenerteller** [comment]
Habe den Verbraucher im Portal nochmal gelöscht und neu angelegt weil ich mit dem Zähler rumgespielt hatte und mir daher einen verbrauch von über 70KW in einer minute eingefangen hatte.
Und was soll ich sagen...jetzt kommt gar nix mehr an.

Hat dieser Logeintrag was zu bedeuten?

2022-12-06 12:43:34,620 WARN [Timer-0] d.a.s.c.MeterReportingSwitch [MeterReportingSwitch.java:170] F-19071974-000000000001-00: Meter not set. 

In einem anderen Fred steht, dass das daran liegt, dass man keinen Schalter konfiguriert hat. Habe ich aber. Typ "Immer an".
Das sieht man auch hier in den SEMP Daten, die sich der Home Manager abholt

![image](https://user-images.githubusercontent.com/119921647/205904764-bb3175ea-cb19-4123-acc8-373eefcae7a0.png)


So langsam verstehe ich gar nix mehr und zuverlässig arbeitet das nicht gerade.

**Comment by Doenerteller** [comment]
ARGHHHHHHHH was ist denn jetzt los? Auf einmal sind Daten da und auch ein Tagesverbrauch?!?!?
Das ist ja alles komplett nicht nachvollziehbar.

![image](https://user-images.githubusercontent.com/119921647/205905337-3875f7dd-0cfc-4bbe-b08d-90884dabf813.png)

**Comment by Doenerteller** [comment]
Falls hier überhaupt jemand mitliest.... es tut sich was, aber die Ergenbnisse sind mal locker 1,5 Stunden hinterher (nein, ich erhebe keinen Anspruch auf nahezu realtime updates, das wäre utopisch). Dann ist es ja kein Wundern dass man denkt, es sei kaputt.
Es ist gerade 17:40.

![image](https://user-images.githubusercontent.com/119921647/205970154-ff49d09f-3bd8-4e83-9fc5-761ea9e7b2e1.png)

Und anscheinend wird auch der Tagesverbrauch nicht sonderlich regelmäßig ermittelt. Eingangs lief das Ding ja schon über 7 Stunden und  der Verbrauch lag noch immer bei Null.

Oder ich hab jetzt unbewußt irgendwas gemacht, dass es "angestossen" hat. Keine Ahnung...

**Comment by skull-77** [comment]
Hi, hat sich das bei dir eigentlich irgendwann gebessert? Bei mir dauert es auch häufig gut 1-2 Std, bis da was aktuelles kommt. Da hab ich dann schon meine Zweifel, dass die Steuerung bedarfsgerecht erfolgt.

**Comment by camueller** [comment]
Sunny Home Manager (macht Steuerung) und das Sunny Portal sind zwei verschiedene Systeme. Bei mir hängt das Portal auch meist 1-2 Stunden hinterher.

**Comment by skull-77** [comment]
Ok, dank dir! Dann passt es ja, auch wenn ich nicht nachvollziehen kann, weshalb es da zu so großen Latezen kommt. 
Ein paar Infos müsste der Home Manager aus der Cloud aber trotzdem bekommen. Zumindest mal das Wetter und die Erwartungswerte. Aber das braucht er natürlich nicht laufend.
